ncm-symlink/src/main/perl/symlink.pod

```diff
@@ -28,92 +25,148 @@ Target definition and contextual variable value may also contain expression inte
+object by the symlink or define the symlink after renaming the object. The nlist keys and values can be:
+
+=over
+
+=item *
+
+key: the existing object type. Valid values are: C<all>, C<dir>, C<dirempty>, C<file>, C<link>, C<none>.  C<dirempty> means an empty directory only, C<dir> means any directory.
+C<all> and C<none> are mutually exclusive but can be combined with other object types to define the extension to use when renaming a given object type or to prevent/enable replacement for a specific object type. 
+
+=item *
 
-    "/software/components/symlink/links" = 
-      list(nlist(
-             "name",    "/usr/local",
-             "target",   "/lal_prod/{ostype}",
-             "exists",    true
-            )
-      );
+value: action applying to the object type. Can be C<yes> (replacement of the object by the symlink allowed), C<no> (replacement of the object by the symlink disabled) or any other
+string. In this latter case, replacement of the object by the symlink is enabled after renaming the object by appending the string to its name. The value can also be empty: see
+below. Note that non empty directories are B<always> renamed before defining the symlink (a default extension, C<.ncm-symlink_saved>, is used).
 
-=head1 DEPENDENCIES
+=back
 
-None.
+=back
 
-=head1 BUGS
+C<replace> option allows a lot of flexibility in specifying what should be done in case of conflict with an existing object. It implements the following advanced features:
 
-None known.
+=over
 
-=head1 AUTHOR
+=item *
 
-Michel Jouvin <jouvin@lal.in2p3.fr>
+C<none=extension> can be used to establish a default rename extension without enabling replacement. This extension will be used with object types for which replacement is enabled.
 
-=head1 MAINTAINER
+=item *
 
-Michel Jouvin <jouvin@lal.in2p3.fr>
+Action can be empty. If a default rename extension was defined with C<none=extension>, the object will be renamed before defining the symlink. Else it is interpreted as C<yes>.
 
-=head1 VERSION
+=back
 
-1.3.2
+=head2 /software/components/symlink/context
+
+A list of contextual variables to use in target definitions.  Each entry is a key/value pair with the variable name
+as the key. The value can contain a command output, as link target definition (see above), with the command to execute
+enclosed between a pair of C<@@>. Contextual variables are global. They are evaluated once, before starting to define
+symlinks.
+
+
+=head2 /software/components/symlink/options
+
+A list of global options used as default for all links creation/deletion. Supported options are the same as options supported in the link definition (see above), 
+with the exception of C<delete>.
+
+=head1 EXAMPLES
 
-=head1 SEE ALSO
+    # Define global variable osdir so that it can be use to define symlink targets
+    "/software/components/symlink/context" = {
+      append(nlist(
+               "name",    "ostype",
+               "value",   "@@uname@@",
```

Comment:
no clearer wording needed, the example makes it very clear. maybe simialr to the way you introduce the variable with `{variable}`, you can do command `@@command@@`?